test/sharing_test.coffee

```diff
@@ -0,0 +1,481 @@
+should    = require('chai').Should()
+_         = require 'lodash'
+sinon     = require 'sinon'
+helpers   = require './helpers'
+rewire    = require 'rewire'
+urlHelper = require 'cozy-url-sdk'
+httpMocks = require 'node-mocks-http'
+events    = require 'events'
+
+sharing      = rewire "#{helpers.prefix}server/controllers/sharing"
+remoteAccess = require "#{helpers.prefix}server/lib/remote_access"
+
+client = helpers.getClient()
+client.setBasicAuth "home", "token"
+
+Client = require('request-json').JsonClient
+clientDS = new Client urlHelper.dataSystem.url()
+
+
+describe 'sharing unit tests', ->
+
+    before helpers.startApp
+
+    after  helpers.stopApp
+
+    # correct objects
+    sharing_request =
+        desc        : 'New mission order'
+        rules       : [{id: 7, docType: 'event'}]
+        sharerUrl   : 'm@mi6.cozy.uk'
+        recipientUrl: 'james-bond@mi6.cozy.uk'
+        preToken    : 'agent-007'
+        shareID     : 18
+
+
+    describe 'createSharing inner module', ->
+
+        createSharing = sharing.__get__ "createSharing"
+
+        it 'When a sharing document is created', (done) ->
+            createSharing sharing_request, (err, docInfo) =>
+                should.not.exist err
+                @docInfo = docInfo
+                should.exist @docInfo._id
+                done()
+
+        it 'then a new document is inserted in the database', (done) ->
+            clientDS.get "data/#{@docInfo._id}", (err, res) ->
+                doc = JSON.parse(res.body)
+                doc.desc.should.equal         sharing_request.desc
+                doc.rules.should.deep.equal   sharing_request.rules
+                doc.sharerUrl.should.equal    sharing_request.sharerUrl
+                doc.recipientUrl.should.equal sharing_request.recipientUrl
+                doc.preToken.should.equal     sharing_request.preToken
+                doc.shareID.should.equal      sharing_request.shareID
+            done()
+
+
+    describe 'revokeFromSharer inner module', ->
+
+        revokeFromSharer = sharing.__get__ 'revokeFromSharer'
+
+        # create a phony access for the sharing added above
+        #before (done) ->
+            #clientDS.post "data/",
+                #{docType: 'access', shareID: sharing_request.shareID},
+                #(err, res, doc) ->
+                    #console.log "DEBUG: doc: " + JSON.stringify doc
+            #done()
+
+        it 'When a sharing is revoked by a sharer'
```

Comment:
I commented out or didn't write some tests because they cannot be run (automatically) until the data-system is up-to-date and I wanted to keep a trace as some sort of reminder. I will change them.